src/NHibernate/AdoNet/OracleDataClientBatchingBatcher.cs

```diff
@@ -125,21 +125,32 @@ protected override void DoExecuteBatch(IDbCommand ps)
 				// this value is not a part of the ADO.NET API.
 				// It's and ODP implementation, so it is being set by reflection
 				SetArrayBindCount(arraySize);
-				int rowsAffected;
 				try
 				{
-					rowsAffected = _currentBatch.ExecuteNonQuery();
+					int rowsAffected;
+					try
+					{
+						rowsAffected = _currentBatch.ExecuteNonQuery();
+					}
+					catch (DbException e)
+					{
+						// Message from DbException could be useful
+						var additionalMessage = string.IsNullOrEmpty(e.Message) ? "." : string.Concat(": ", e.Message);
```

Comment:
@oskarb this is how it was already implemeted. The real change here is additional try-finally around the execution block. I think it shall go to 4.1